tests/core_test.py

```diff
@@ -258,6 +259,23 @@ def test_jvp_2(self):
     assert d2_sin(0.0) == 0.0
     assert d3_sin(0.0) == -1.0
 
+  def test_reference_cycles(self):
+    def f(x):
+      return x.sum()
+
+    fn = partial(linearize, f)
+    params = np.zeros([])
+    fn(params)
```

Comment:
I think the first `fn(params); gc.collect()` here can be elided. I called `fn` twice in my example code to ensure that I understood correctly how JIT impacted the circular reference behavior (i.e. verifying that the circular references occurred as a result of tracing, and that JITted functions wouldn't incur this cost).

Reply:
Thanks!